### ISSUE TITLE
null value for module does not set the default value

### ISSUE THREAD

**Issue by simonweil** [hashicorp/terraform#29858]
### Terraform Version

```
$ terraform version
Terraform v1.0.10
on darwin_amd64
+ provider registry.terraform.io/hashicorp/time v0.7.2
```

### Terraform Configuration Files

test.tf:
```terraform
module "foo" {
  source = "./foo"
  disable = null
}
```

foo/main.tf:
```terraform
variable "disable" {
  type = bool
  default = false
}

resource "time_sleep" "wait_30_seconds" {
  count = var.disable ? 1 : 0

  create_duration = "30s"
}
```

### Debug Output

$ TF_LOG=trace terraform apply
[https://gist.github.com/simonweil/0fa27b0d0e3eb9622180b74521fee74c](https://gist.github.com/simonweil/0fa27b0d0e3eb9622180b74521fee74c)

### Expected Behavior
As described [in the docs](https://www.terraform.io/docs/language/expressions/types.html#null), when `null` is set to the module's variable, the variable will get the default value of `false` and the resource is not created.

> null: a value that represents absence or omission. If you set an argument of a resource or **module to null, Terraform behaves as though you had completely omitted it — it will use the argument's default value if it has one**, or ...

### Actual Behavior
The apply fails and with an error that the value of the variable is `null`...

```terraform
$ terraform apply
╷
│ Error: Null condition
│
│   on foo/main.tf line 12, in resource "time_sleep" "wait_30_seconds":
│   12:   count = var.disable ? 1 : 0
│     ├────────────────
│     │ var.disable is null
│
│ The condition value is null. Conditions must either be true or false.
```

### Steps to Reproduce
1. `terraform init`
2. `terraform apply`

**Comment by jbardin** [comment]
Hi @simonweil,

Thanks for filing the issue. We just closed the duplicate issue which you can see here #24142.
The fix if planned for the v1.1 release.

Thanks!

**Comment by simonweil** [comment]
Great there is a fix on the way!
Is there a timelines for 1.1?
